core/app/models/spree/promotion_handler/coupon.rb

```diff
@@ -8,7 +8,7 @@ class Coupon
 
       def initialize(order)
         @order = order
-        @coupon_code = order.coupon_code && order.coupon_code.downcase
+        @coupon_code = order.coupon_code(&:downcase)
```

Comment:
I'm not sure about this. It's both not really related to this change and I think it works only since `Spree::Order#coupon_code` is a method but it won't work if you apply this syntax to a variable, for example:

```console
> coupon_code = 'TEst'
> coupon_code(&:downcase)
NoMethodError: undefined method `coupon_code' for main:Object


> def coupon_code; "TEst"; end
> coupon_code(&:downcase)
"TEst"
```

Also, I think it's not working as expected (the string is not downcased).